compiler/src/iree/compiler/Codegen/Utils/VectorOpUtils.cpp

```diff
@@ -52,6 +52,10 @@ VectorContractOpInfo::inferFromIndexingMaps(ArrayRef<AffineMap> maps) {
   opInfo.lhsKDim = *maps[0].getResultPosition(getAffineDimExpr(k, ctx));
   opInfo.rhsKDim = *maps[1].getResultPosition(getAffineDimExpr(k, ctx));
 
+  opInfo.lhsUnitDims = maps[0].getBroadcastDims();
+  opInfo.rhsUnitDims = maps[1].getBroadcastDims();
+  opInfo.accUnitDims = maps[2].getBroadcastDims();
```

Comment:
Can you add a check at the top of this function for `isProjectedPermutation(/*allowZeroInResults=*/true)` for each map? I don't think `inferContractionDims` checks for that.